### PR TITLE
fix: deactivate cosmic plugin when not in a COSMIC session

### LINKED ISSUE #169: Noticable lag introduced in the last commit
Today I noticed that everytime that I invoke the launcher and type something in it, it will not show any results for around 1 second and then it populates the results. 

There are a few panics in the logs that mention cctk top level:
```
pop-os gnome-shell[35603]: thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', /build/pop-launcher-lRDvef/pop-launcher-1.2.2~1675445713~22.04~0b8e385/vendor/cosmic-client-toolkit/src/toplevel_info.rs:38:14

```

### PR DESCRIPTION
**Pull request by mmstick**
This change will ensure that the cosmic plugin is deactivated if the launcher is not in a COSMIC session. It checks if `XDG_CURRENT_DESKTOP` contains `COSMIC` to do so.

Fixes #169

### CODE REVIEW

**Review by wash2 — APPROVED** [review]

**Comment by jacobgkau** [comment]
I haven't been able to recreate https://github.com/pop-os/launcher/issues/169 (tested in GNOME on Pop!_OS with COSMIC Epoch installed, Pop!_OS without COSMIC Epoch installed, as well as on Ubuntu since the reporter mentioned Apport.)

This fix doesn't cause any problems in GNOME, but it seems to prevent the launcher from showing or searching open windows on COSMIC Epoch. That would be a regression? (Even though we're not shipping COSMIC Epoch yet.)

**Comment by hojjatabdollahi** [comment]
>  as well as on Ubuntu since the reporter mentioned Apport.)
I'm on Pop. I had to mask the Apport service to get rid of the lag.
 
I just tested it again. It doesn't happen on Pop XOrg, but happens on Pop Wayland. 
I have been using Wayland for so long I forgot to mention it, my apologies.

**Comment by mmstick** [comment]
@jacobgkau What's your output for `XDG_CURRENT_DESKTOP` in COSMIC? It should be `pop:COSMIC`. Unless the launcher is running in an environment where this is not passed down to it.

**Comment by mmstick** [comment]
Also yeah I've been having this issue in the Wayland session of Pop.

**Review by wash2 — APPROVED** [review]
Ok seems to be working in Pop, Pop on Wayland, and COSMIC for me now

File: plugins/src/cosmic_toplevel/mod.rs [hunk 1]
```diff
@@ -24,7 +24,14 @@ use self::toplevel_handler::{toplevel_handler, ToplevelAction, ToplevelEvent};
 pub async fn main() {
     tracing::info!("starting cosmic-toplevel");
 
-    let (mut app, mut toplevel_rx) = App::new(async_stdout());
+    let mut tx = async_stdout();
+
+    if session_is_cosmic() {
```
[the remark this answers is not in the document]
Comment (mmstick) [reply plugins/src/cosmic_toplevel/mod.rs]: That's the second `!` I've missed in the last week.

**Review by jacobgkau — APPROVED** [review]
062a0a8 has no delay in GNOME (X11), GNOME (Wayland), or COSMIC. Open windows show in all three.

<details>
<summary>Regression testing passed in GNOME (X11):</summary>

- [X] All windows on all workspaces appear on launch
- [X] Choosing an app on another workspace moves workspaces and focus to that app
- [X] Launching an application works
- [X] Typing text and then removing it will re-show those windows
- [X] Search works for applications and windows
- [X] Search works for GNOME settings panels
- [X] Search for "Extensions". There should be only one entry.
- [X] The overlay hint correctly highlights the selected window
- [X] Open windows are sorted above applications (e.g. "firefox")
- [X] t: executes a command in a terminal
- [X] : executes a command in sh
- [X] = calculates an equation
- [X] Search results are as expected:
    - `cal` returns Calendar and Calculator before Color
    - `pops` returns Popsicle first
    - `shop` returns the Pop!_Shop first
</details>

**Pull request merged**